File: src/main/java/com/example/demo/service/UserService.java
package com.example.demo.service;

import java.util.List;

import com.example.demo.dto.UserDto;

/**
 * User Service Interface
 * 
 * @author Suresh Chinnaswamy
 *
 */
public interface UserService {

	UserDto getUserById(Integer userId);

	void saveUser(UserDto userDto);

	List<UserDto> getAllUsers();

	Integer authenticate(UserDto userDto);
}

File: src/main/resources/static/js/app/UserService.js
'use strict'

angular.module('demo.services', []).factory('UserService',
		[ "$http", "CONSTANTS", function($http, CONSTANTS) {
			var service = {};
			var userId;
			service.getUserById = function(userId) {
				var url = CONSTANTS.getUserByIdUrl + userId;
				return $http.get(url);
			}
			
			
			service.authenticate = function(userDto) {
				return $http.post('user/authenticate', userDto);
			}
			
			service.setUserId = function(data) {
			   userId = data;
			}
			
			service.getUserId = function() {
				return userId;
			}
			
			return service;
		} ]);